torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -2296,37 +2296,97 @@ class IrParser {
           });
     }
 
-    /*
-    // TODO: Enable view in parser by detecting non-alias view operation
     {
-      std::array<const char*, kNumViewSize> View = {
-          "aten::view(Tensor(a) self, int[] size) -> Tensor(a)",
-          "aten::reshape(Tensor(a) self, int[] shape) -> Tensor(a)"};
-      for (auto signature : View) {
+      auto ptr_op = getOperatorForLiteral(
+          "prim::view_copy(Tensor(a) self, int[] size) -> Tensor(a)");
+      REGISTER_PARSE_RULE(
+          ptr_op,
+          {
+            auto self_value = node->inputs()[0];
+            MemoryFormat format;
+            std::list<Val*> list_val;
+            std::tie(format, list_val) = getConsistentValues(
+                MemoryFormat::Contiguous(), value_map[self_value->unique()]);
+            auto self = list_val.front()->as<TensorView>();
+            list_val.pop_front();
+
+            auto self_type = self_value->type()->cast<c10::TensorType>();
+            TORCH_INTERNAL_ASSERT(self_type != nullptr);
+            auto self_sizes = getTensorSizes(self_type);
+
+            auto view_sizes =
+                constant_as<c10::List<int64_t>>(node->input(1));
+            TORCH_INTERNAL_ASSERT(
+                view_sizes.has_value(), "The size parameter is required.");
+
+            auto output = view(self, self_sizes, view_sizes->vec());
+            value_map.emplace(node->output()->unique(), output);
+          },
+          nullptr,
```

Comment:
this is where we can add the logic to reject fusion of `view_copy` where shape has `-1`.

e.g. https://github.com/csarofeen/pytorch/blob/cef28edbdb0e7ccd38f095addc61bfdf5a884bd5/torch/csrc/jit/codegen/cuda/parser.cpp#L1857-L1862